internal/test/scorecard/tests.go

```diff
@@ -72,6 +76,61 @@ func CryostatCRTest(bundle *apimanifests.Bundle, namespace string, openShiftCert
 	return *r
 }
 
+func CryostatConfigChangeTest(bundle *apimanifests.Bundle, namespace string, openShiftCertManager bool) scapiv1alpha3.TestResult {
+	tr := newTestResources(CryostatConfigChangeTestName)
+	r := tr.TestResult
+
+	err := setupCRTestResources(tr, openShiftCertManager)
+	if err != nil {
+		return fail(*r, fmt.Sprintf("failed to set up %s test: %s", CryostatConfigChangeTestName, err.Error()))
+	}
+
+	// Create a default Cryostat CR with default empty dir
+	cr := newCryostatCR(CryostatConfigChangeTestName, namespace, !tr.OpenShift)
+	cr.Spec.StorageOptions = &operatorv1beta1.StorageConfiguration{
+		EmptyDir: &operatorv1beta1.EmptyDirConfig{
+			Enabled: true,
+		},
+	}
+
+	_, err = createAndWaitTillCryostatAvailable(cr, tr)
+	if err != nil {
+		return fail(*r, fmt.Sprintf("%s test failed: %s", CryostatConfigChangeTestName, err.Error()))
+	}
+
+	// Switch Cryostat CR to PVC for redeployment
+	ctx, cancel := context.WithTimeout(context.Background(), testTimeout)
+	defer cancel()
+	client := tr.Client
+
+	cr, err = client.OperatorCRDs().Cryostats(namespace).Get(ctx, CryostatConfigChangeTestName)
+	if err != nil {
+		return fail(*r, fmt.Sprintf("failed to get Cryostat CR: %s", err.Error()))
+	}
+	cr.Spec.StorageOptions = &operatorv1beta1.StorageConfiguration{
+		PVC: &operatorv1beta1.PersistentVolumeClaimConfig{
+			Spec: &corev1.PersistentVolumeClaimSpec{
+				StorageClassName: nil,
+				Resources: corev1.ResourceRequirements{
+					Requests: corev1.ResourceList{
+						corev1.ResourceStorage: resource.MustParse("1Gi"),
+					},
+				},
+			},
+		},
+	}
+
+	// Wait for redeployment of Cryostat CR
+	err = updateAndWaitTillCryostatAvailable(cr, tr)
+	if err != nil {
+		return fail(*r, fmt.Sprintf("Cryostat redeployment did not become available: %s", err.Error()))
+	}
+	r.Log += "Cryostat deployment has successfully updated to pvc"
+	defer cleanupCryostat(r, tr.Client, CryostatConfigChangeTestName, namespace)
```

Comment:
Would it make sense to call defer after 

```bash
_, err = createAndWaitTillCryostatAvailable(cr, tr)
```

That way, if any errors occur, the CR is cleaned up regardless?